tests/functional/metadata/mongodb/putObject.spec.js

```diff
@@ -0,0 +1,442 @@
+const async = require('async');
+const assert = require('assert');
+const werelogs = require('werelogs');
+const { MongoMemoryReplSet } = require('mongodb-memory-server');
+const { errors, versioning } = require('../../../../index');
+const logger = new werelogs.Logger('MongoClientInterface', 'debug', 'debug');
+const BucketInfo = require('../../../../lib/models/BucketInfo');
+const MetadataWrapper =
+    require('../../../../lib/storage/metadata/MetadataWrapper');
+const { DbPrefixes } = versioning.VersioningConstants;
+
+const VID_SEP = '\0';
+const IMPL_NAME = 'mongodb';
+const DB_NAME = 'metadata';
+const BUCKET_NAME = 'test-bucket';
+const OBJECT_NAME = 'test-object';
+const BUCKET_VFORMAT_V0 = 'v0';
+const VERSION_ID = '98451712418844999999RG001  22019.0';
+
+const mongoserver = new MongoMemoryReplSet({
+    debug: false,
+    instanceOpts: [
+        { port: 27021 },
+    ],
+    replSet: {
+        name: 'rs0',
+        count: 1,
+        DB_NAME,
+        storageEngine: 'ephemeralForTest',
+    },
+});
+
+function unescape(obj) {
+    return JSON.parse(JSON.stringify(obj).
+                      replace(/\uFF04/g, '$').
+                      replace(/\uFF0E/g, '.'));
+}
+
+function formatMasterKeyV0(key) {
+    return `${key}`;
+}
+
+function formatMasterKeyV1(key) {
+    return `${DbPrefixes.Master}${key}`;
+}
+
+function formatVersionKeyV0(key, versionId) {
+    return `${key}${VID_SEP}${versionId}`;
+}
+
+function formatVersionKeyV1(key, versionId) {
+    return `${DbPrefixes.Version}${formatVersionKeyV0(key, versionId)}`;
+}
```

Comment:
I was surprised these are not available in source :shrug: maybe there's some factorization we can do after this PR

Reply:
yep i agree